src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -383,22 +383,69 @@ private void changeLabels() {
      * Adds a style class to the listview which changes its background to contain a loading spinning gif
      */
     @Override
-    protected void addPanelLoadingIndication() {
+    protected void showLoadingIndicator() {
         logger.info("Preparing to add panel loading indication");
         listView.getStyleClass().add("listview-loading");
-
-        // Remove the items in the issue list because they are not relevant to the filter anymore.
-        // This also makes the listview background visible, since we intend to show a loading indicator on the
-        // background.
-        listView.setItems(null);
     }
 
     /**
-     * Removes the style class that was added in addPanelLoadingIndicator() from the listview.
+     * Removes the style class that was added in showLoadingIndicator() from the listview.
      */
     @Override
-    protected void removePanelLoadingIndication() {
+    protected void hideLoadingIndicator() {
         logger.info("Preparing to remove panel loading indication");
         listView.getStyleClass().removeIf(cssClass -> cssClass.equals("listview-loading"));
     }
+
+    @Override
+    protected void startLoadingAnimationIfApplicable(PrimaryRepoOpeningEvent e) {
+        if (isIndicatorApplicable()) startLoadingAnimation();
+    }
+
+    @Override
+    protected void stopLoadingAnimationIfApplicable(PrimaryRepoOpenedEvent e) {
+        if (isIndicatorApplicable()) stopLoadingAnimation();
+    }
+
+    @Override
+    protected void startLoadingAnimationIfApplicable(ApplyingFilterEvent e) {
+        if (e.panel == this) startLoadingAnimation();
+    }
+
+    @Override
+    protected void stopLoadingAnimationIfApplicable(AppliedFilterEvent e) {
+        if (e.panel == this) stopLoadingAnimation();
+    }
+
+    @Override
+    protected void startLoadingAnimation() {
+        setTranslucentCellFactory();
+        showLoadingIndicator();
+    }
+
+    @Override
+    protected void stopLoadingAnimation() {
+        hideLoadingIndicator();
+    }
+
+    private void setTranslucentCellFactory() {
+        if (getElementsList() == null) return;
+        final HashSet<Integer> issuesWithNewComments
+                = updateIssueCommentCounts(Qualifier.hasUpdatedQualifier(getCurrentFilterExpression()));
+        listView.setCellFactory(list -> {
+            ListPanelCell cell = new ListPanelCell(this, panelIndex, issuesWithNewComments);
+            cell.setStyle(cell.getStyle() + "-fx-opacity: 40%;");
+            return cell;
+        });
+    }
+
+    /**
+     * Determines whether the loading indicator should show
+     *
+     */
+    private boolean isIndicatorApplicable() {
+        HashSet<String> allReposInFilterExpr =
+                Qualifier.getMetaQualifierContent(getCurrentFilterExpression(), QualifierType.REPO);
+        return allReposInFilterExpr.isEmpty();
+    }
```

Comment:
Can give clear behaviour explanation for this method in the comment? i.e. why is the indicator only applicable when there are no repos in filter.